codegen/java/src/main/kotlin/io/spine/protodata/codegen/java/Expression.kt

```diff
@@ -0,0 +1,386 @@
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+@file:JvmName("Expressions")
+
+package io.spine.protodata.codegen.java
+
+import com.google.common.collect.ImmutableList
+import com.google.common.collect.ImmutableMap
+import com.google.protobuf.ByteString
+import io.spine.protobuf.TypeConverter
+import io.spine.protodata.Field
+import io.spine.protodata.Field.CardinalityCase
+import io.spine.protodata.Field.CardinalityCase.LIST
+import io.spine.protodata.Field.CardinalityCase.MAP
+import io.spine.protodata.Field.CardinalityCase.SINGLE
+import io.spine.protodata.FieldName
+import io.spine.protodata.camelCase
+import kotlin.reflect.KClass
+
+private const val COPY_OF = "copyOf"
+private const val OF = "of"
+
+private val immutableListClass = ClassName(ImmutableList::class)
+private val immutableMapClass = ClassName(ImmutableMap::class)
+
+/**
+ * A piece of Java code which yields a value.
+ */
+public sealed class Expression(private val code: String) {
+
+    /**
+     * Prints this Java expression.
+     */
+    public fun toCode(): String = code
+
+    final override fun toString(): String = toCode()
+
+    /**
+     * Obtains an `Expression` which wraps this `Expression`'s value in a `com.google.protobuf.Any`.
+     */
+    public fun packToAny(): Expression {
+        val type = ClassName(TypeConverter::class)
+        return type.call("toAny", arguments = listOf(this))
+    }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+        other as Expression
+        if (code != other.code) return false
+        return true
+    }
+
+    override fun hashCode(): Int =
+        code.hashCode()
+}
+
+/**
+ * An literal `null` expression.
+ */
+public object Null : Expression("null")
+
+/**
+ * A string literal.
+ *
+ * Represented as the same value as the given string, wrapped in quotation marks. No extra character
+ * escaping is perfomed.
+ */
+public class LiteralString(value: String) : Expression("\"$value\"")
+
+private val byteStringClass = ByteString::class.qualifiedName!!
+
+/**
+ * An expression which yields the given byte string.
+ */
+public class LiteralBytes(bytes: ByteString) : Expression(
+    "$byteStringClass.copyFrom(new byte[]{${bytes.toByteArray().joinToString()}})"
+)
+
+/**
+ * An expression represented by the given string.
+ *
+ * No extra processing is done upon the given code.
+ */
+public class Literal(value: Any) : Expression(value.toString())
+
+/**
+ * A fully qualified Java class name.
+ *
+ * In Java, a class name is not a valid expression. Use one of the methods of this class to create
+ * an expression from this class name.
+ */
+public data class ClassName
+/**
+ * Creates a class name from the given FQN.
+ */
+constructor(private val className: String) {
```

Comment:
Is this layout intended?

Reply:
Yes, primary constructors in Kotlin are declared outside the class body. I was not able to find an official recommendation on this one. 

This alignment is what IDEA does when declaring the primary constructor on a new line.

Some people prefer keeping the `constructor` keyword on the same line with the class name like so:
```kotlin
public class Expression constructor(
    private val className: String
) // ...
```
However, given the presence of a doc, which separates the constructor from the class name, we have no choice but to write it on a new line.
Also, I'd argue, this layout more clear when there is a `constructor` keyword, like in this file.

Without the keyword, the list of constructor params should probably just be written in a column, or on the same line as the class name in some simple cases. For example:
```kotlin
public class Expression(
    private val className: String
) {
    // Class body.
}

public data class Email(val value: String) // No class body, no alignment-based highlithing is required.
```